test/library/packages/LinearAlgebra/performance/me/me-perf.chpl

```diff
@@ -0,0 +1,401 @@
+use Time;
+use Math;
+use LinearAlgebra;
+use Random;
+
+config var m=4,iters=10,correctness=false;
+config var iterLimit = max(int);
+config var timeLimit = 10;
+
+config type eltType = real;
+
+const nbytes = numBytes(eltType);
+
+
+proc test_expm() {
+  const Space = {0..<m, 0..<m};
+  var t: Timer;
+
+  if !correctness {
+    writeln('=========================================');
+    writeln('  Matrix Exponential Performance Test');
+    writeln('=========================================');
+    writeln('iters : ', iters);
+    writeln('m     : ', m);
+    writeln('numLocales     : ', numLocales);
+    writeln('iterLimit     : ', iterLimit);
+    writeln('timeLimit     : ', timeLimit);
+    writeln('MB    : ', (nbytes*m*m) / 10**6);
+    writeln();
+  }
+
+  {
+    // Pade3, Pade5, Pade7, Pade9, Pade13 cases
+
+    var b = [0.01, 0.1, 0.5, 1.0, 10.0];
+    const pade = [3, 5, 7, 9, 13];
+
+    for i in b.domain do {
+      var X: [Space] eltType;
+      setDiag(X, 0, 1);
+      var Y = X * b[i];
+
+      var M: [X.domain] eltType;
+
+      var curIter = 0;
+      t.start();
+      do {
+        for 1..iters {
+          expm(Y);
+        }
+        curIter += iters;
+      } while (curIter <= iterLimit && t.elapsed() <= timeLimit);
+      t.stop();
+
+      if !correctness then
+        writeln('LinearAlgebra.expm_pade' + (pade[i]: string)
+                              + ': ' + ((t.elapsed()/curIter): string));
+      else
+        writeln("");
+
+      t.clear();
+    }
+    writeln("");
+  }
+
+
+
+  {
+    // Input Matrices of Format
+    // [1.0 10^n 0.0]
+    // [0.0 -1.0 0.0]
+    // [0.0  0.0 1.0]
+
+    var b = [1.0, 10.0, 100.0, 1000.0, 10000.0, 100000.0,
+                                  1000000.0, 10000000.0, 100000000.0];
+
+    for i in b.domain do {
+      var X: [Space] eltType;
+      setDiag(X, 0, 1);
+      X(1,1) = -1.0;
+      X(0,1) = b[i];
+
+      var M: [X.domain] eltType;
```

Comment:
are we using `M`?

Reply:
it appears in others cases as well

Reply:
I initially used `M` to hold returned values of the test methods in question and used it to print it for correctness. I removed it now. I'll push the code now.